govcd/adminvdc.go

```diff
@@ -424,3 +424,171 @@ func (vdc *AdminVdc) UpdateStorageProfile(storageProfileId string, storageProfil
+		}
+	}
+
+	href := vdc.AdminVdc.HREF + "/vdcStorageProfiles"
+
+	var updateStorageProfile = types.UpdateVdcStorageProfiles{
+		Xmlns:                types.XMLNamespaceVCloud,
+		Name:                 storageProfile.Name,
+		Description:          "",
+		RemoveStorageProfile: storageProfile,
+	}
+
+	task, err := vdc.client.ExecuteTaskRequest(href, http.MethodPost,
+		types.MimeUpdateVdcStorageProfiles, "error removing VDC storage profile: %s", &updateStorageProfile)
+	if err != nil {
+		return Task{}, fmt.Errorf("cannot remove VDC storage profile, error: %s", err)
+	}
+
+	return task, nil
+}
+
+// RemoveStorageProfileWait removes a storege profile from a VDC and returns a refreshed VDC or an error
+func (vdc *AdminVdc) RemoveStorageProfileWait(storageProfileName string) error {
+	task, err := vdc.RemoveStorageProfile(storageProfileName)
+	if err != nil {
+		return err
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return err
+	}
+	return vdc.Refresh()
+}
+
+// SetDefaultStorageProfile sets a given storage profile as default
+// This operation will automatically unset the previous default storage profile.
+func (vdc *AdminVdc) SetDefaultStorageProfile(storageProfileName string) error {
+	if vdc.client.VCDHREF.String() == "" {
+		return fmt.Errorf("cannot Set VDC default storage profile: VCD HREF is unset")
+	}
+
+	var storageProfile *types.Reference
+	for _, sp := range vdc.AdminVdc.VdcStorageProfiles.VdcStorageProfile {
+		if sp.Name == storageProfileName {
+			storageProfile = sp
+		}
+	}
+	if storageProfile == nil {
+		return fmt.Errorf("cannot set VDC default storage profile: storage profile '%s' not found in VDC", storageProfileName)
+	}
+
+	vdcStorageProfileDetails, err := vdc.client.GetStorageProfileByHref(storageProfile.HREF)
+	if err != nil {
+		return fmt.Errorf("cannot retrieve VDC storage profile '%s' details: %s", storageProfileName, err)
+	}
+	_, err = vdc.UpdateStorageProfile(extractUuid(storageProfile.HREF), &types.AdminVdcStorageProfile{
+		Name:    storageProfile.Name,
+		Units:   "MB",
+		Limit:   vdcStorageProfileDetails.Limit,
+		Default: true,
+		Enabled: takeBoolPointer(true),
+		ProviderVdcStorageProfile: &types.Reference{
+			HREF: vdcStorageProfileDetails.ProviderVdcStorageProfile.HREF,
+		},
+	},
+	)
+	if err != nil {
+		return fmt.Errorf("cannot Set VDC default storage profile '%s': %s", storageProfileName, err)
+	}
+	return vdc.Refresh()
+}
+
+// GetDefaultStorageProfileReference finds the default storage profile for the VDC
+func (adminVdc *AdminVdc) GetDefaultStorageProfileReference() (*types.Reference, error) {
+	var defaultSp *types.Reference
+	if adminVdc.AdminVdc.VdcStorageProfiles == nil || adminVdc.AdminVdc.VdcStorageProfiles.VdcStorageProfile == nil {
+		return nil, fmt.Errorf("no storage profiles found in VDC %s", adminVdc.AdminVdc.Name)
+	}
+	for _, sp := range adminVdc.AdminVdc.VdcStorageProfiles.VdcStorageProfile {
+		fullSp, err := adminVdc.client.GetStorageProfileByHref(sp.HREF)
```

Comment:
what here means `full` ?

Reply:
"full", as opposed to the simple reference that we had until that point.